File: src/lecture/02_Images/SaveImage/SaveImage.cpp
/*****************************************************************************************************
 * Lecture sample code.
 * Open an image as grayscale image and save it to a file.
 *****************************************************************************************************
 * Author: Marc Hensel, http://www.haw-hamburg.de/marc-hensel
 * Project: https://github.com/MarcOnTheMoon/imaging_learners/
 * Copyright: 2023, Marc Hensel
 * Version: 2023.09.09
 * License: CC BY-NC-SA 4.0, see https://creativecommons.org/licenses/by-nc-sa/4.0/deed.en
 *****************************************************************************************************/

/* Include files */
#include <opencv2/opencv.hpp>

/* Main function */
int main()
{
	// Load image from file
	cv::Mat image = cv::imread("D:/Docks.jpg", cv::IMREAD_GRAYSCALE);

	// Display image in named window
	cv::imshow("Image", image);

	// Save image to file
	cv::imwrite("D:/Docks_gray.jpg", image);

	// Wait for keypress and terminate
	cv::waitKey(0);
	return 0;
}

File: src/python/lecture/02_Images/SaveImage.py
"""
Open an image as grayscale image and save it to a file.

@author: Marc Hensel
@contact: http://www.haw-hamburg.de/marc-hensel

@copyright: 2025, Marc Hensel
@version: 2025.03.30
@license: CC BY-NC-SA 4.0, see https://creativecommons.org/licenses/by-nc-sa/4.0/deed.en
"""

import cv2

# Load image from file
image = cv2.imread('D:/Docks.jpg', cv2.IMREAD_GRAYSCALE)

# Display image in named window
cv2.imshow('Image [Press any key to quit]', image)

# Save image to file
cv2.imwrite('D:/Docks_gray.jpg', image)

# Wait for keypress
cv2.waitKey(0)

# Close all named windows
cv2.destroyAllWindows()